pkg/controller/kieapp/webconsole.go

```diff
@@ -0,0 +1,141 @@
+package kieapp
+
+//go:generate go run -mod=vendor ./defaults/.packr/packr.go
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/gobuffalo/packr/v2"
+	"strconv"
+	"strings"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	TargetNamespace   = "openshift-console"
+	ConsoleYAMLSample = "ConsoleYAMLSample"
+)
+
+type CustomResourceDefinition struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+func CreateConsoleYAMLSamples(r *Reconciler) {
+	log.Info("Loading CR yaml samples.")
+	contents, names, err := loadFilesOnlyWithBox("cryamlsamples")
+	if err != nil {
+		log.Error(err, "Failed to load cryamlsamples")
+		return
+	}
+
+	resMap, err := applyMultipleWebConsoleYamls(contents, r)
+	if err != nil {
+		log.Error(err, "cryamlsamples not successfully applied")
+		return
+	}
+
+	for k, v := range resMap {
+		log.Info("yaml file: %s %s ", names[k], v)
+	}
+}
+
+func applyMultipleWebConsoleYamls(yamlString []string, r *Reconciler) (map[int]string, error) {
+	if yamlString == nil || len(yamlString) == 0 {
+		return nil, errors.New("Empty yaml list")
+	}
+	resMap := make(map[int]string)
+	for idx, yamlStr := range yamlString {
+		_, err := create(yamlStr, r)
+		if err != nil {
+			resMap[idx] = err.Error()
+		} else {
+			resMap[idx] = "Applied"
+		}
+	}
+	return resMap, nil
+}
+
+func create(yamlStr string, r *Reconciler) (bool, error) {
+	obj := &CustomResourceDefinition{}
+	err := yaml.Unmarshal([]byte(yamlStr), obj)
+	if err != nil {
+		return false, err
+	}
+
+	kind := obj.Annotations["consolekind"]
+	if kind != ConsoleYAMLSample {
+		return false, errors.New("Unrecognized Kind or Kind is missing.")
+	}
+
+	snippetStr := obj.Annotations["consolesnippet"]
+	var snippet bool = false
+	if snippetStr != "" {
+		tmp, err := strconv.ParseBool(snippetStr)
+		if err != nil {
+			return false, errors.New("Unable to parse snippet to boolean.")
+		}
+		snippet = tmp
+	}
+	title, _ := obj.Annotations["consoletitle"]
+	desc, _ := obj.Annotations["consoledesc"]
+	name, _ := obj.Annotations["consolename"]
+
+	yamlSample := &consolev1.ConsoleYAMLSample{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: TargetNamespace,
+		},
+		Spec: consolev1.ConsoleYAMLSampleSpec{
+			TargetResource: v1.TypeMeta{
+				APIVersion: obj.TypeMeta.APIVersion,
+				Kind:       obj.TypeMeta.Kind,
+			},
+			Title:       consolev1.ConsoleYAMLSampleTitle(title),
+			Description: consolev1.ConsoleYAMLSampleDescription(desc),
+			YAML:        consolev1.ConsoleYAMLSampleYAML(yamlStr),
+			Snippet:     snippet,
+		},
+	}
+	err = r.Service.Create(context.TODO(), yamlSample)
```

Comment:
Now thinking about what can be generic, this line won't be portable. I think we should make a simple util function that takes the CR and returns a `ConsoleYAMLSample`, and let the operator create it.

Reply:
modified